src/lib.rs

```diff
@@ -11,13 +11,42 @@ pub mod objects;
 pub mod version_prefix;
 pub mod whitespace_remover;
 
+/// Decodes a blueprint string and compares that parsed json matches
+/// after we decode and then re-encode it.
+pub fn roundtrip_blueprint_test(blueprint: &str) {
+    let bp2 =
+        BlueprintCodec::encode_string(&BlueprintCodec::decode_string(blueprint).unwrap()).unwrap();
+    let json_v1 = decode_to_json_value(blueprint.as_bytes()).unwrap();
+    let json_v2 = decode_to_json_value(bp2.as_bytes()).unwrap();
+
+    if json_v1 != json_v2 {
+        let mut w = std::fs::File::create("output1.json").unwrap();
+        write!(w, "{:#}", json_v1).unwrap();
+
+        let mut w = std::fs::File::create("output2.json").unwrap();
+        write!(w, "{:#}", json_v2).unwrap();
+        panic!("Mismatched output found, compare output of output1.json and output2.json");
```

Comment:
Writing to the filesystem like this feels super messy, particularly in the case that more than one test fails. Later failures will simply overwrite earlier failures. Worse, as tests are run in parallel, the final value of `output1.json` may not correspond to the final value of `output2.json` at all. Feel free to emit to the console--the test runner handles synchronization in that case--but I don't like this implementation at all.

Reply:
I implemented it like that as printing to stdout where a small portion of a huge blueprint differs, it's basically useless. 

Looks like https://crates.io/crates/assert-json-diff is nearly what would be wanted.